startup.py

```diff
@@ -200,6 +193,17 @@ def _icon_from_executable(self, code_name):
     def _find_software(self):
         """Find executables in the default install locations."""
 
+        # This dictionary defines a list of executable template strings for each
+        # of the supported operating systems. The templates are used for both
+        # globbing and regex matches by replacing the named format placeholders
+        # with an appropriate glob or regex string.
+        self.EXECUTABLE_TEMPLATES = {
```

Comment:
I'm not a big fan of overwritting the `EXECUTABLE_TEMPLATES` variable. Upper case variables are constants by conventions.

Could you instead create a new local variable and make sure it's used in the rest of the function?